aptos-move/block-executor/src/scheduler.rs

```diff
@@ -186,15 +186,21 @@ impl ValidationStatus {
     }
 }
 
-pub struct Scheduler {
-    /// Number of txns to execute, immutable.
-    num_txns: TxnIndex,
+impl Default for ValidationStatus {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+pub struct Scheduler<P: SchedulerProvider> {
+    provider: Arc<P>,
 
     /// An index i maps to indices of other transactions that depend on transaction i, i.e. they
     /// should be re-executed once transaction i's next incarnation finishes.
-    txn_dependency: Vec<CachePadded<Mutex<Vec<TxnIndex>>>>,
+    txn_dependency: P::TxnDependencyCollection,
```

Comment:
This should move to the provider.

Reply:
Why not here? Leave it here, so we can make the provider immutable.